### PR TITLE
Simplified test suite that runs on Windows and MacOS

### PR DESCRIPTION
**Pull request by sergv**
As discussed in https://github.com/haskell/alex/issues/84, it's good idea to add test suite before messing with code generation that would interplay with C preprocessor. This test suite also runs Linux builds to help establish whether it's working at all as opposed to not working due to some Windows-related issue.

What's missing is that tests run with the C preprocessor that comes bundled with GHC. As outlined deep in https://github.com/haskell/alex/issues/84, some issues were present with cpphs preprocessor that Agda used at the time. 

It seems test suite could be parameterised by the C preprocessor, but it seems it would be best done on a test suite that uses testing framework like `tasty`. That would bring other benefits like being able to run specific tests through command-line selectors. Perhaps the makefile-based test suite is overdue to be updated?

### CODE REVIEW

File: .github/workflows/haskell-ci-simple.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    - name: Test
+      run: |
+        cd alex-*/
+        export ALEX=$(cabal list-bin alex)
+        cabal run --enable-tests alex:test:tests
+
+    - name: Haddock
+      run: |
+        cd alex-*/
+        cabal haddock all
```
Comment (andreasabel) [inline .github/workflows/haskell-ci-simple.yml:77]: This invocation does not seem to work:
> No documentation was generated as this package does not contain a library.

(See the other workflow for the correct haddock invocation.)

**Review by andreasabel — CHANGES_REQUESTED** [review]
Great!

One change: the haddock invocation does not work yet.

File: .github/workflows/haskell-ci-simple.yml (hunk 1 above)
Comment (sergv) [reply .github/workflows/haskell-ci-simple.yml]: This reminds me, wouldn't it be a better structure to have alex package define library and use it from the executable?

File: .github/workflows/haskell-ci-simple.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
           - "8.10.7"
           - "9.0.2"
           - "9.2.7"
-          - "9.4.5"
+          - "9.4.4"
```
Comment (andreasabel) [inline .github/workflows/haskell-ci-simple.yml:27]: Shouldn't 9.4.5 work with the latest haskell-setup?  @haskell-actions/setup@v2  (note migration to a new organization)

**Review by andreasabel — CHANGES_REQUESTED** [review]
Thanks for fixing the haddock step!
Could you have a look whether 9.4.5 works with the latest setup action?
`haskell-actions/setup@v2`

**Comment by sergv** [comment]
@andreasabel Looks like haskell-actions/setup@v2 works with GHC 9.4.5.

**Review by andreasabel — APPROVED** [review]

**Pull request merged**